src/hooks/useTheme/useTheme.ts

```diff
@@ -29,8 +31,11 @@ const { lightTheme: defaultTheme } = themes
  *
  * @returns {Theme} The current theme.
  */
-export const useTheme = (): Theme => {
+export const useTheme = (): { theme: Theme, antTheme: ThemeConfig } => {
   const theme = useContext(ThemeContext)
 
-  return useMemo(() => theme ?? defaultTheme, [theme])
+  const themeWithFallback = useMemo(() => theme ?? defaultTheme, [theme])
+  const antTheme = useMemo(() => generateAntTheme(themeWithFallback), [themeWithFallback])
+
+  return { theme: themeWithFallback, antTheme }
```

Comment:
It feels odd for the `useTheme` to return an object holding a `theme` property. I'd expect the hook to return the actual theme (as in current implementation).


Also providing the underlying antd theme as our public interface doesn't feel like a sound choice. Can we provide an additional hook that returns the antdTheme so that only when the it is specifically needed you'd use the new hook instead? This would also make the change non-breaking (not that it is a problem since we are at v0, but this would still require some effort on all the existing codebase using it so I think it must be thought thouroughly)

Reply:
Sure, I have re-implemented the feature as a new hook and changed the specifications of the PR